mesonbuild/compilers/c.py

```diff
@@ -704,10 +706,12 @@ def has_function(self, funcname, prefix, env, extra_args=None, dependencies=None
              * We would always check for this, but we get false positives on
              * MSYS2 if we do. Their toolchain is broken, but we can at least
              * give them a workaround. */
-            #if {no_includes:d}
+            #if {no_includes:d} && !{func_is_builtin:d}
```

Comment:
I think we should try if the idea with implicit includes allows us to avoid this.

Reply:
I think we should do that in a separate PR since it has a large potential for a yak-shave.

Reply:
If this is the only place where we would need that, i'm ok with splitting that part of into a separate PR.